tests/product/tests/data/k8s/k8s_test_case.py

```diff
@@ -0,0 +1,40 @@
+"""
+This module provides k8s test case definition
+"""
+
+from dataclasses import dataclass, astuple
+
+
+@dataclass
+class FileTestCase:
+    """
+    Represents k8s nodes test case
+    """
+
+    rule_tag: str
+    node_hostname: str
+    resource_name: str
+    expected: str
+
+    def __iter__(self):
+        return iter(astuple(self))
+
+    def __len__(self):
+        return len(astuple(self))
```

Comment:
You're correct; we don't need `__iter__` and `__len__` methods for the data class since it's solely used for storing test data, and additional manipulations are not expected; removed